scipy/stats/_distn_infrastructure.py

```diff
@@ -2853,6 +2854,108 @@ def entropy(pk, qk=None, base=None, axis=0):
     return S
 
 
+def differential_entropy(
+    values: np.ndarray,
+    *,
+    window_length: int = 1,
+    base: Optional[float] = None,
+    axis: int = 0,
+) -> np.ndarray:
+    r"""Given a sample of a distribution, calculate the differential entropy.
+
+    This routine uses the Vasicek estimator of the differential entropy. Given
+    a sorted random sample :math:`X_1, \ldots X_n` this is defined as:
+
+    .. math::
+        \frac{1}{n}\sum_1^n \log\left[ \frac{n}{2m} (X_{i+m} - X_{i-m}) \right]
+
+    where :math:`m` is the window length parameter, :math:`X_{i} = X_1` if
+    :math:`i < 1` and :math:`X_{i} = X_n` if :math:`i > n`.
+
+    Parameters
+    ----------
+    values : sequence
+        Samples of the (continuous) distribution.
+    window_length : int, optional
+        Window length for computing Vasicek estimate. Must be an integer
+        between 1 and half of the sample size.
+    base : float, optional
+        The logarithmic base to use, defaults to ``e`` (natural logarithm).
+    axis: int, optional
+        The axis along which the differential entropy is calculated.
+        Default is 0.
+
+    Returns
+    -------
+    entropy : float
+        The calculated differential entropy.
+
+    Examples
+    --------
+
+    >>> from scipy.stats import differential_entropy, norm
+
+    Entropy of a standard normal distribution:
+
+    >>> random_state = np.random.RandomState(0)
+    >>> values = random_state.standard_normal(100)
+    >>> differential_entropy(values)
+    1.1220441777259473
+
+    Compare with the true entropy:
+
+    >>> float(norm.entropy())
+    1.4189385332046727
+
+    Increasing the window length can improve the accuracy:
+
+    >>> differential_entropy(values, window_length=8)
+    1.3494014875503249
+
+    References
+    ----------
+    .. [1] Vasicek, O. (1976). A test for normality based on sample entropy.
+           Journal of the Royal Statistical Society:
+           Series B (Methodological), 38(1), 54-59.
+
+    """
+    values = asarray(values)
+    values = np.moveaxis(values, axis, 0)
+
+    if not 2 <= 2 * window_length < len(values):
```

Comment:
If there's not a sensible reason for a negative base, could error for that, too.

Reply:
The `entropy` function does not have that check either. I think that they should both share the same behaviour.

Reply:
Sure, but maybe that is an argument that `entropy` _should_ have the check, too.  For example, if `entropy` had a certain bug, certainly we wouldn't intentionally include that bug here. (Not that any changes to `entropy` should hold up this PR, though.)

It is not very important, so I will resolve this if another maintainer answers the [next comment](https://github.com/scipy/scipy/pull/13631#discussion_r587943987) without also suggesting for this change.